File: OnlineShop.API/Models/ProductSelection.cs
namespace OnlineShop.API.Models
{
    public class ProductSelection
    {
        public int IdProduct { get; set; }
        public string Name { get; set; }
        public decimal Price { get; set; }
        public string PhotoUrl { get; set; }
        public int Quantity { get; set; }
    }
}

File: OnlineShop-SPA/src/app/_models/ProductSelection.ts
import { Cart } from './Cart';

export class ProductSelection {

  constructor(public cart: Cart,
              public idProduct?: number,
              public name?: string,
              public price?: number,
              public  photoURL?: string,
              private quantityValue?: number){}

  get quantity(){
    return this.quantityValue;
  }

  set quantity(newQuantity: number){
    this.quantityValue = newQuantity;
    this.cart.update();
  }
}
